SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,859 @@
+#include "ExpectedGeneratedSchemaFileContents.h"
+
+#include "Utils/SchemaDatabase.h"
 
 #include "CoreMinimal.h"
+#include "HAL/PlatformFilemanager.h"
+#include "Misc/FileHelper.h"
 
 #define SCHEMA_GENERATOR_TEST(TestName) \
 	TEST(SpatialGDKEditor, SchemaGenerator, TestName)
 
+namespace
+{
+const FString gSchemaOutputFolder = FPaths::Combine(FSpatialGDKServicesModule::GetSpatialOSDirectory(), TEXT("Tests/"));
+const FString gSchemaDatabaseFileName = TEXT("Spatial/Tests/SchemaDatabase");
+const FString gDatabaseOutputFile = TEXT("/Game/Spatial/Tests/SchemaDatabase");
+
+TArray<FString> LoadSchemaFileForClassToStringArray(const FString& SchemaOutputFolder, const UClass* CurrentClass)
+{
+	FString SchemaFileFolder = TEXT("");
+
+	if (!CurrentClass->IsChildOf<AActor>())
+	{
+		SchemaFileFolder = TEXT("Subobjects");
+	}
+
+	TArray<FString> FileContent;
+	FFileHelper::LoadFileToStringArray(FileContent, *FPaths::SetExtension(FPaths::Combine(FPaths::Combine(SchemaOutputFolder, SchemaFileFolder), CurrentClass->GetName()), TEXT(".schema")));
+
+	return FileContent;
+}
+
+struct NamesAndIds
+{
+	TArray<FString> Names;
+	TArray<int32> Ids;
+};
+
+NamesAndIds ParseAvailableNamesAndIdsFromSchemaFile(const FString& SchemaOutputFolder, const UClass* CurrentClass)
+{
+	const TArray<FString> LoadedSchema = LoadSchemaFileForClassToStringArray(SchemaOutputFolder, CurrentClass);
+	NamesAndIds ParsedNamesAndIds;
+
+	for (const auto& SchemaLine : LoadedSchema)
+	{
+		if (SchemaLine.Contains(TEXT("\tid =")))
+		{
+			TArray<FString> OutArray;
+			SchemaLine.ParseIntoArrayWS(OutArray);
+			FString ParsedId = OutArray.Last().LeftChop(1);
+
+			if (ParsedId.IsNumeric())
+			{
+				ParsedNamesAndIds.Ids.Push(FCString::Atoi(*ParsedId));
+			}
+		}
+		else if (SchemaLine.Contains(TEXT("component ")))
+		{
+			TArray<FString> OutArray;
+			SchemaLine.ParseIntoArrayWS(OutArray);
+			if ((OutArray.Num() > 1) && (OutArray[0].Compare(TEXT("component")) == 0))
+			{
+				ParsedNamesAndIds.Names.Push(OutArray[1]);
+			}
+		}
+	}
+
+	return ParsedNamesAndIds;
+}
+
+bool TestEqualDatabaseEntryAndSchemaFile(const UClass* CurrentClass, const FString& SchemaOutputFolder, const USchemaDatabase* SchemaDatabase)
+{
+	NamesAndIds ParsedNamesAndIds = ParseAvailableNamesAndIdsFromSchemaFile(SchemaOutputFolder, CurrentClass);
+
+	if (CurrentClass->IsChildOf<AActor>())
+	{
+		const FActorSchemaData* ActorData = SchemaDatabase->ActorClassPathToSchema.Find(CurrentClass->GetPathName());
+		if (ActorData == nullptr)
+		{
+			return false;
+		}
+		else
+		{
+			if (ParsedNamesAndIds.Names.Num() != 1)
+			{
+				return false;
+			}
+
+			if (ActorData->GeneratedSchemaName.Compare(ParsedNamesAndIds.Names[0]) != 0)
+			{
+				return false;
+			}
+
+			// TODO(Alex): how to test this?
+			//ActorData->SubobjectData;
+
+			for (int i = 0; i < ParsedNamesAndIds.Ids.Num(); ++i)
+			{
+				if (ActorData->SchemaComponents[i] != ParsedNamesAndIds.Ids[i])
+				{
+					return false;
+				}
+			}
```

Comment:
How come we this fails if there is more than one component name in the schema?

Reply:
If it's an Actor, than we expect to see just one component name